### ISSUE TITLE
rake db:seed is throwing an error

### ISSUE THREAD

**Issue by ArushiSinghal** [EBWiki/EBWiki#1312]
Hello

While setting up an EBWiki development environment.
The  ``` rake --trace db:seed ``` is  giving me following error
```
Creating Articles...
rake aborted!
NoMethodError: undefined method `ansi_code' for nil:NilClass
/home/seema/EBWIKI/EBWiki/app/models/article.rb:68:in `full_address'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/geocoder-1.4.5/lib/geocoder/stores/base.rb:88:in `do_lookup'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/geocoder-1.4.5/lib/geocoder/stores/active_record.rb:300:in `geocode'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/activesupport-4.2.9/lib/active_support/callbacks.rb:432:in `block in make_lambda'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/activesupport-4.2.9/lib/active_support/callbacks.rb:145:in `block in halting_and_conditional'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/activesupport-4.2.9/lib/active_support/callbacks.rb:504:in `block in call'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/activesupport-4.2.9/lib/active_support/callbacks.rb:504:in `each'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/activesupport-4.2.9/lib/active_support/callbacks.rb:504:in `call'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/activesupport-4.2.9/lib/active_support/callbacks.rb:92:in `__run_callbacks__'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/activesupport-4.2.9/lib/active_support/callbacks.rb:778:in `_run_save_callbacks'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/activerecord-4.2.9/lib/active_record/callbacks.rb:302:in `create_or_update'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/activerecord-4.2.9/lib/active_record/persistence.rb:120:in `save'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/bullet-5.7.0/lib/bullet/active_record42.rb:47:in `save_with_bullet'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/activerecord-4.2.9/lib/active_record/validations.rb:37:in `save'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/activerecord-4.2.9/lib/active_record/attribute_methods/dirty.rb:21:in `save'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/activerecord-4.2.9/lib/active_record/transactions.rb:286:in `block (2 levels) in save'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/activerecord-4.2.9/lib/active_record/transactions.rb:351:in `block in with_transaction_returning_status'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/activerecord-4.2.9/lib/active_record/connection_adapters/abstract/database_statements.rb:213:in `block in transaction'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/activerecord-4.2.9/lib/active_record/connection_adapters/abstract/transaction.rb:184:in `within_new_transaction'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/activerecord-4.2.9/lib/active_record/connection_adapters/abstract/database_statements.rb:213:in `transaction'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/activerecord-4.2.9/lib/active_record/transactions.rb:220:in `transaction'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/activerecord-4.2.9/lib/active_record/transactions.rb:348:in `with_transaction_returning_status'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/activerecord-4.2.9/lib/active_record/transactions.rb:286:in `block in save'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/activerecord-4.2.9/lib/active_record/transactions.rb:301:in `rollback_active_record_state!'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/activerecord-4.2.9/lib/active_record/transactions.rb:285:in `save'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/activerecord-4.2.9/lib/active_record/persistence.rb:34:in `create'
/home/seema/EBWIKI/EBWiki/db/seeds.rb:333:in `block in <top (required)>'
/home/seema/EBWIKI/EBWiki/db/seeds.rb:332:in `each'
/home/seema/EBWIKI/EBWiki/db/seeds.rb:332:in `<top (required)>'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/activesupport-4.2.9/lib/active_support/dependencies.rb:268:in `load'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/activesupport-4.2.9/lib/active_support/dependencies.rb:268:in `block in load'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/activesupport-4.2.9/lib/active_support/dependencies.rb:240:in `load_dependency'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/activesupport-4.2.9/lib/active_support/dependencies.rb:268:in `load'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/railties-4.2.9/lib/rails/engine.rb:547:in `load_seed'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/activerecord-4.2.9/lib/active_record/tasks/database_tasks.rb:253:in `load_seed'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/activerecord-4.2.9/lib/active_record/railties/databases.rake:173:in `block (2 levels) in <top (required)>'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/rake-12.3.0/lib/rake/task.rb:251:in `block in execute'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/rake-12.3.0/lib/rake/task.rb:251:in `each'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/rake-12.3.0/lib/rake/task.rb:251:in `execute'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/rake-12.3.0/lib/rake/task.rb:195:in `block in invoke_with_call_chain'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/2.4.0/monitor.rb:214:in `mon_synchronize'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/rake-12.3.0/lib/rake/task.rb:188:in `invoke_with_call_chain'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/rake-12.3.0/lib/rake/task.rb:181:in `invoke'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/rake-12.3.0/lib/rake/application.rb:160:in `invoke_task'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/rake-12.3.0/lib/rake/application.rb:116:in `block (2 levels) in top_level'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/rake-12.3.0/lib/rake/application.rb:116:in `each'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/rake-12.3.0/lib/rake/application.rb:116:in `block in top_level'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/rake-12.3.0/lib/rake/application.rb:125:in `run_with_threads'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/rake-12.3.0/lib/rake/application.rb:110:in `top_level'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/rake-12.3.0/lib/rake/application.rb:83:in `block in run'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/rake-12.3.0/lib/rake/application.rb:186:in `standard_exception_handling'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/rake-12.3.0/lib/rake/application.rb:80:in `run'
/home/seema/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/rake-12.3.0/exe/rake:27:in `<top (required)>'
/home/seema/.rbenv/versions/2.4.1/bin/rake:22:in `load'
/home/seema/.rbenv/versions/2.4.1/bin/rake:22:in `<main>'
Tasks: TOP => db:seed
```

**Comment by rlgreen91** [comment]
Which of the articles is causing this error?

**Comment by ArushiSinghal** [comment]
'State' article
```
NoMethodError: undefined method `ansi_code' for nil:NilClass
/FILE/EBWiki/app/models/article.rb:68:in `full_address'
```

**Comment by rlgreen91** [comment]
That doesn't make sense.  If you look at the seeds file, there is no "state" article - instead, we create an array of hashes, each hash representing an article, and run a loop to process each hash.  Which one of those is causing the error?

**Comment by trystant** [comment]
To be clear, line 68 looks like the following:
```
     { ansi_code: 'BC', iso: 'CA-BC', name: 'British Columbia' },
```

Is this the line in question?

**Comment by ArushiSinghal** [comment]
@trystant 
The line 68, which is giving an error is this 
https://github.com/EBWiki/EBWiki/blob/b9e60d5e7d9f76171e155ab683eac52922d1c0f4/app/models/article.rb#L67-L69

The  error is   
````NoMethodError: undefined method `ansi_code' for nil:NilClass```

**Comment by rlgreen91** [comment]
Yes, but which of the articles is causing the error? If you start the Rails console, and create the states as listed, and then try to create each article, which one causes that error?

**Comment by rlgreen91** [comment]
Hey @ArushiSinghal just following up on this - were you able to get this to work?

**Comment by harikarajavaram** [comment]
I am also getting the same error. @rlgreen91 could you please tell me the solution to this error?

**Comment by rlgreen91** [comment]
@harikarajavaram if you look at `db/seeds.rb`, you can see that we run a loop to create the articles. If you run through that loop manually, which one of those articles causes the error?

**Comment by simransinghal** [comment]
@harikarajavaram you successfully installed all the dependencies?

**Comment by trystant** [comment]
I've updated the `db/seeds.rb` file so that the states are created & references found. This should resolve this seeding issue.

**Comment by rlgreen91** [comment]
Addressed in #1492
